synapse/rest/client/v1/room.py

```diff
@@ -294,7 +294,7 @@ def on_GET(self, request):
             # In both cases we call the auth function, as that has the side
             # effect of logging who issued this request if an access token was
             # provided.
-            if server:
+            if server or self.hs.config.require_auth_for_room_directory:
```

Comment:
I think this is the wrong place for this check. Specifically, it blocks local users from requesting the directory from remote servers. It does nothing to stop users on remote servers from requesting our directory, which uses a different endpoint, and is what I think you're after.

Try here: https://github.com/matrix-org/synapse/blob/b20602eeb87d5cba05ae6f1b808c4295cdbc4e3f/synapse/federation/transport/server.py#L574